src/interpreter/kernel/exec.js

```diff
@@ -170,6 +172,32 @@ export function executeStackFrame(
     return res;
   }
 
+  function getMemoryOffset(instruction) {
+    if (instruction.namedArgs && instruction.namedArgs.offset) {
+      const offset = instruction.namedArgs.offset;
+      if (offset < 0) {
+        throw new RuntimeError("offset must be positive");
+      }
+      if (offset > 0xffffffff) {
+        throw new RuntimeError(
+          "offset must be less than or equal to 0xffffffff"
+        );
+      }
+      return offset;
+    } else {
+      return 0;
```

Comment:
IRC I had a constant `NULL = 0x0`, we can use it for every memory operations if it can make things clearer?